posts/2020-07-31-Rust-1.45.2.md

```diff
@@ -0,0 +1,48 @@
+---
+layout: post
+title: "Announcing Rust 1.45.2"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.45.2. Rust is a
+programming language that is empowering everyone to build reliable and
+efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.45.2 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.45.2][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1452-2020-07-31
+
+## What's in 1.45.2 stable
+
+1.45.2 contains two fixes, one to 1.45.1 and the other to 1.45.0.
+
+## `#[track_caller]` on trait objects
+
+Trait objects and `#[track_caller]` could lead to unsoundness. `#[track_caller]`
+is not yet stable on 1.45. However, the standard library makes use of this on
+some traits for better error messages. Trait objects of `SliceIndex`, `Index`,
+and `IndexMut` are affected by this bug.
+
+## Bindings in tuple-patterns
```

Comment:
The bug actually manifested in a slice pattern.

Reply:
Well, the backported fix was for tuple patterns, and the regression was in a slice pattern. So we're going back to the status quo from 1.45.0, and I guess that ICE will re-emerge for tuples.

Reply:
Hm, I think the bug is something like "slice patterns in tuple patterns" or something like that. I'm not sure how to pinpoint it.

Reply:
Oh, because `Option::Some(_)` is a tuple variant? OK, that makes sense, and I just checked that there's no issue for slice patterns alone.

Reply:
Maybe "Nested bindings in tuple patterns"?

Reply:
The condition leads me to believe it needs to be a tuple struct (or tuple enum variant, I guess) specifically with a `..` pattern bound to a variable in it. Saying all of that in the heading seems too verbose, though. Perhaps "Tuple patterns containing bindings to `..`"? That also feels rather long.